pyanaconda/modules/payloads/payload/live_image/installation.py

```diff
@@ -41,28 +41,39 @@
 class DownloadImageTask(Task):
     """Task to download an image."""
 
-    def __init__(self, configuration: LiveImageConfigurationData, image_path):
+    def __init__(self, configuration: LiveImageConfigurationData, download_path):
         """Create a new task.
 
         :param configuration: a configuration of a remote image
         :type configuration: an instance of LiveImageConfigurationData
-        :param str image_path: a path to the downloaded image
+        :param str download_path: a path to the downloaded image
         """
         super().__init__()
         self._url = configuration.url
         self._proxy = configuration.proxy
         self._ssl_verify = configuration.ssl_verification_enabled
-        self._image_path = image_path
+        self._download_path = download_path
 
     @property
     def name(self):
         """Name of the task."""
         return "Download an image"
 
     def run(self):
-        """Run the task."""
+        """Run the task.
+
+        If the image is local, we return its local location. Otherwise,
+        the image is downloaded at the specified download location and
+        we return that one.
+
+        :return: a path to the image
+        """
         log.info("Downloading the image...")
 
+        if self._url.startswith("file://"):
```

Comment:
There are already other `download` methods in this class, so I didn't want to introduce new ones. Also, I didn't want to hide the testing for the `file://` prefix into a function since we remove that prefix two lines later. It would be confusing.